### PR TITLE
Llama2

### PR DESCRIPTION
**Pull request by philschmid**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: huggingface/pytorch/tgi/docker/0.9.3/py3/cu118/Dockerfile.gpu [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+# Install server
+COPY proto proto
+COPY server server
+COPY server/Makefile server/Makefile
+RUN cd server && \
+    make gen-server && \
+    pip install -r requirements.txt && \
+    pip install ".[bnb, accelerate]" --no-cache-dir
+
```
Comment (xyang16) [inline huggingface/pytorch/tgi/docker/0.9.3/py3/cu118/Dockerfile.gpu:180]:
```suggestion
RUN rm -r proto server
```

Comment (xyang16) [reply huggingface/pytorch/tgi/docker/0.9.3/py3/cu118/Dockerfile.gpu:180]: Add this to clean the source code after installing.

File: huggingface/pytorch/tgi/docker/0.9.3/py3/cu118/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+# Copy build artifacts from custom kernels builder
+COPY --from=custom-kernels-builder /usr/src/build/lib.linux-x86_64-cpython-39 /opt/conda/lib/python3.9/site-packages
+
+# Copy builds artifacts from vllm builder
+COPY --from=vllm-builder /usr/src/vllm/build/lib.linux-x86_64-cpython-39 /opt/conda/lib/python3.9/site-packages
+
+# Install flash-attention dependencies
+RUN pip install einops --no-cache-dir
+
```
Comment (xyang16) [inline huggingface/pytorch/tgi/docker/0.9.3/py3/cu118/Dockerfile.gpu:171]:
```suggestion
RUN rm -r transformers/examples
```

Comment (xyang16) [reply huggingface/pytorch/tgi/docker/0.9.3/py3/cu118/Dockerfile.gpu:171]: Clean the examples code.

File: huggingface/pytorch/tgi/docker/0.9.3/py3/cu118/Dockerfile.gpu [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip && \
+    unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ && \
+    cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance && \
+    chmod +x /usr/local/bin/testOSSCompliance && \
+    chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh && \
+    ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} python && \
+    rm -rf ${HOME_DIR}/oss_compliance*
+RUN curl -o /root/THIRD-PARTY-LICENSES https://publish.djl.ai/dlc-licenses/huggingface/tgi-0.8.2/THIRD-PARTY-LICENSES
+
+# Final image
```
Comment (xyang16) [inline huggingface/pytorch/tgi/docker/0.9.3/py3/cu118/Dockerfile.gpu:209]:
```suggestion
```

File: huggingface/pytorch/tgi/docker/0.9.3/py3/cu118/Dockerfile.gpu [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+FROM base
```
Comment (xyang16) [inline huggingface/pytorch/tgi/docker/0.9.3/py3/cu118/Dockerfile.gpu:210]:
```suggestion
```

Comment (xyang16) [reply huggingface/pytorch/tgi/docker/0.9.3/py3/cu118/Dockerfile.gpu:210]: We need the sagemaker image only

**Review by xyang16 — APPROVED** [review]

**Comment by nth-attempt** [comment]
Thanks @philschmid for this PR. @xyang16 will this image be available in ECR as well?  thanks!

**Comment by xyang16** [comment]
@nth-attempt Yes it will be available in ECR once it's released.

**Comment by monuminu** [comment]
waiting for it soon :)

**Comment by xyang16** [comment]
@ashivadi 

The directory structure is like this:

```
$tree 
.
├── text-generation-inference
└── ── llm-hosting-container
```

Working directory is `text-generation-inference`

**Comment by ashivadi** [comment]
> @ashivadi
> 
> The directory structure is like this:
> 
> ```
> $tree 
> .
> ├── text-generation-inference
> └── ── llm-hosting-container
> ```
> 
> Working directory is `text-generation-inference`

Thanks, updated here: https://github.com/ashivadi/llama2-with-tgi-sagemaker

**Comment by nth-attempt** [comment]
Is the image available in EC2 yet?

**Comment by xyang16** [comment]
@nth-attempt The target release date is by Monday next week.

**Comment by xyang16** [comment]
@nth-attempt TGI 0.9.3 released.

**Comment by nth-attempt** [comment]
Thank you!! https://github.com/aws/deep-learning-containers/blob/master/available_images.md#huggingface-text-generation-inference-containers

**Pull request merged**